### ISSUE TITLE
docs for Set#add aren't true

### ISSUE THREAD

**Issue by trinarytree** [dart-lang/sdk#25713]
Currently it says
"Adds value into the set. Returns true if value was added to the set.

If value already exists, the set is not changed and false is returned."

The 1st sentence says that it unconditionally adds value into the set.
The 2nd sentence says that it returns true if the thing it just said always happens happens.
Therefore it must always return true.
The hypothesis of the 3rd sentence is always true: you passed value, therefore it exists (yes, I know what was meant, but it's not what was said).
The conclusion of the 3rd sentence now disagrees with our earlier conclusion that this method must return true.

Analogy:
```dart
/// Definitely eats a sandwich, no question.  Returns true if it eats a sandwich.
/// If this method is called, no sandwich is eaten and false is returned.
bool eatSandwich();
```

Can we change this to something more literally correct?  E.g.
"Adds value into the set provided it's not already there and returns whether it was added."
or if you want to get super exact
"Adds value into the set if and only if it's not already there and returns whether it was added."

**Comment by floitschG** [comment]
Thanks.
Updated the documentation in 381c12df4ade7cfe67aedd391a1896072a6fc799.

**Comment by trinarytree** [comment]
The 1st sentence is still not literally true.  This method doesn't unconditionally add the value to the set.  Compare to the exactness of the analogous [Java docs](http://docs.oracle.com/javase/7/docs/api/java/util/Set.html#add(E)).
I know the next few sentences are correct, but sentences 1 and 3 shouldn't disagree with each other.
Those couple of extra words go a long way toward building trust in the docs.
What about
"Adds [value] to the set provided an equivalent value is not already in it.
Returns whether [value] was added."

**Comment by floitschG** [comment]
I agree that the documentation could be more formal, but I don't think it's needed. There is a trade-off between easy parsing/readability and formal correctness.
The goal of the documentation is to provide a fast way (for example in a hover of the IDE) to see what the method does. Corner cases should be treated in additional paragraphs.
It is generally assumed that adding an existing entry to a set won't increase the size of the set, so adding that information to the first sentence would just make reading the comment harder. It is handled in additional paragraphs and examples.

Also, we are discussing adding "Advanced" sections to dartdoc. These could then be more formal and precise, and contain even more information that is usually not needed for first-glance reading of the documentation.

To summarize:
- a first sentence that makes it easy to see what the function does.
- more paragraphs that explain details, corner-cases, ...
- (eventually) an advanced-section that would cover all the things that are usually not needed.
